salt/metalk8s/kubernetes/apiserver/kubeconfig.sls

```diff
@@ -4,7 +4,7 @@
 include:
   - metalk8s.internal.m2crypto
 
-{%- set apiserver = 'https://' ~ pillar.metalk8s.api_server.host ~ ':6443' %}
+{%- set apiserver = 'https://' ~ grains['metalk8s']['control_plane_ip'] ~ ':6443' %}
```

Comment:
Why do not use the "proxy" here ?
That mean if for what ever reason local apiserver do not work the `admin.conf` will not work and since we use this kubeconfig everywhere it's not really good IMO

Reply:
Because this conf is perfectly valid to be used from *outside* the node, e.g. an admin copying it to his machine (aren't we even doing that for tests?)

Note how *most* `KubeConfig`s are rewritten to use the local proxy, so the services using those shouldn't be impacted. AFAIK only `salt-master` is using `admin.conf` (which is a bug by itself...).

In practice, not much changes, because in our current deployments the `apiServer.host` value is set to the hostname or IP of the bootstrap node anyway...

Reply:
Not that any *script* which uses `admin.conf` and is *certain* it's running on a host on which the proxy is deployed, could pass `-s`/`--server` to the `kubectl` invocation to use the proxy anyway.

Reply:
Ok but then I think we need to change something in our current restore script because if I well remember we need to contact apiserver before configuring the local one and we basically call `kubeconfig` from apiserver and then use this new admin,conf

Reply:
That wouldn't work today in the first place then, unless the APIserver address was HA...

Reply:
Indeed 
```
.. warning::

    You cannot use the restore script if you do not have High Availability
    apiserver because some information required to reconfigure the others
    nodes are stored in the apiserver.
```
PS: Oh btw you should remove this section in this PR also

Reply:
No I mean, that's not a reasonable requirement in the first place.

Instead, the recover script should take, as an argument, the address of *an* API-server, and use that address when connecting to do whatever it needs to do.

Reply:
With this PR yes indeed but when restore script get writed in our bootstrap config the goal was to have an IP able to hit apiserver, and not only bound to bootstrap IP, either with keepalived or with any other system because configuring all server with the bootstrap IP as apiserver endpoint is really wrong.
But yes now we need to change this

Reply:
My point is that a VIP or LB for API-server has *always* been *optional*, so the restore script *relying* on that is wrong in the first place.

The bug fixed in this PR is the fact that sometimes we relied on this given address to be HA, given the context of #2108 

The restore script should also be updated to support this properly, and I'm happy to open a bug for this, but I'd rather not move this into this PR, since the context is completely different IMHO.

Finally, the lack of CI for the restore operation (#1687) basically *allows* me to merge this without remorse ;-P

Reply:
Ugh, I keep putting comments, but the topic keeps resonating.

What I was thinking, primarily: there *should* be no `admin.conf` in the backup, because it's not at all required to restore a cluster. An `admin.conf` can be generated from the CA data that *does* need to be backed up.

The whole restore process should be constructive, starting from the minimal set of required data as kept in a backup.

Reply:
So, I went to see, and apparently there's not even an `admin.conf` in the backup archive (which is good!)

However, what *is* in there has a really strange structure... We really need to spend some time on that it seems.